src/Sema.zig

```diff
@@ -6526,12 +6528,70 @@ fn zirArrayCat(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Ai
                 sema.analyzeDeclRef(try anon_decl.finish(ty, val))
             else
                 sema.analyzeDeclVal(block, .unneeded, try anon_decl.finish(ty, val));
-        } else {
-            return sema.fail(block, lhs_src, "TODO runtime array_cat", .{});
         }
-    } else {
-        return sema.fail(block, lhs_src, "TODO runtime array_cat", .{});
     }
+
+    // at runtime!
+    const usize_ty = Type.initTag(.usize);
+    const array_ty = if (res_sent) |rs|
+        try Type.Tag.array_sentinel.create(sema.arena, .{ .len = final_len, .elem_type = lhs_info.elem_type, .sentinel = rs })
+    else
+        try Type.Tag.array.create(sema.arena, .{ .len = final_len, .elem_type = lhs_info.elem_type });
+    const ptr_ty = try Type.ptr(sema.arena, .{
+        .pointee_type = array_ty,
+        .@"addrspace" = target_util.defaultAddressSpace(sema.mod.getTarget(), .local),
+    });
+    const alloc = try block.addTy(.alloc, ptr_ty);
+
+    // TODO there is some ownership issue where if one of these is comptime-known, it generates an invalid tv
+    const ptr_to_lhs = if (!is_pointer) try sema.analyzeRef(block, lhs_src, lhs) else lhs;
+    const ptr_to_rhs = if (!is_pointer) try sema.analyzeRef(block, rhs_src, rhs) else rhs;
```

Comment:
Do I have to copy one of these into sema.arena if it is comptime known?
I get invalid int values inside the air array.
![image](https://user-images.githubusercontent.com/58830309/135734813-c8767636-5fd4-4ca5-a68f-d67b2fe865d2.png)

Reply:
I put in some code to handle it, and its still there ...